Frameworks/UIKit/StarboardXaml/ApplicationMain.mm

```diff
@@ -56,14 +61,29 @@ int ApplicationMainStart(const char* principalName,
 
     // Populate Objective C equivalent of activation argument
     if (activationType == ActivationTypeToast) {
-        NSString* toastArgument = Strings::WideToNSString(static_cast<HSTRING>(activationArg));
-        activationArgument = toastArgument;
+        WAAToastNotificationActivatedEventArgs* toastArgument = [WAAToastNotificationActivatedEventArgs createWith:activationArg];
+        NSMutableDictionary* userInput = [NSMutableDictionary new];
+        for (NSString* key in [toastArgument.userInput allKeys]) {
```

Comment:
This one is for the willFinishLaunchingWithOptions: and didFinishLaunchingWithOptions: while the other is for didReceiveToastAction:

---
In reply to: [72317600](https://github.com/Microsoft/WinObjC/pull/637#discussion_r72317600) [](ancestors = 72317600)